samples: net: sockets: net_mgmt: Increase main thread priority

rlubos opened the pull request:
net_mgmt sockets do not implement internal queue but use net_mgmt_event_wait() internally to receive events. As a consequence, in case of events burst, some events may be lost if the receiving thread has lower priority than the net_mgmt thread.

This was visible in the sample, where only DAD event was reported properly, as IPv6 Add/Remove events were triggered in pair with corresponding multicast events (from solicited-node multicast address).

We can mitigate this by increasing main thread priority to be at the same priority as net_mgmt thread, so that the receiving thread (main) has a chance to run in between event reports.

rlubos committed:
samples: net: sockets: net_mgmt: Increase main thread priority

net_mgmt sockets do not implement internal queue but use
net_mgmt_event_wait() internally to receive events. As a consequence,
in case of events burst, some events may be lost if the receiving
thread has lower priority than the net_mgmt thread.

This was visible in the sample, where only DAD event was reported
properly, as IPv6 Add/Remove events were triggered in pair with
corresponding multicast events (from solicited-node multicast
address).

We can mitigate this by increasing main thread priority to be at the
same priority as net_mgmt thread, so that the receiving thread (main)
has a chance to run in between event reports.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

GeorgeCGV commented:
The commit & PR information can be hard to find or notice. Won't it be better to mention that in `config NET_MGMT_EVENT_THREAD` description or to have a separate priority config option?

rlubos committed:
net: lib: sockets: net_mgmt: Add note about thread priorities

Add a note about the thread priority requirements in the help string of
the Kconfig option enabling net_mgmt sockets.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos commented:
> The commit & PR information can be hard to find or notice. Won't it be better to mention that in `config NET_MGMT_EVENT_THREAD` description or to have a separate priority config option?

I've added this information to the help string of the corresponding Kconfig option. We don't really have any other documentation for this feature at the moment.
